app/src/main/java/com/space_intl/chatapp/presentation/ui/chat/adapter/ChatAdapter.kt

```diff
@@ -5,18 +5,29 @@ import android.view.View
 import android.view.ViewGroup
 import android.widget.ImageView
 import android.widget.TextView
-import com.space_intl.chatapp.R
+import androidx.annotation.ColorRes
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
 import com.space_intl.chatapp.common.extensions.setBackgroundTint
 import com.space_intl.chatapp.common.extensions.setTint
-import com.space_intl.chatapp.common.extensions.toFormattedDate
+import com.space_intl.chatapp.common.util.C
+import com.space_intl.chatapp.common.util.S
 import com.space_intl.chatapp.databinding.LayoutMessageItemBinding
-import com.space_intl.chatapp.domain.model.MessageModel
-import com.space_intl.chatapp.presentation.base.adapter.BaseAdapter
+import com.space_intl.chatapp.presentation.base.adapter.CustomItemCallback
+import com.space_intl.chatapp.presentation.ui.chat.model.MessageUIModel
 
-typealias C = R.color
+class ChatAdapter(private val listener: () -> String) :
+    ListAdapter<MessageUIModel, ChatAdapter.ChatViewHolder>(CustomItemCallback<MessageUIModel>()) {
 
-class ChatAdapter(listener: () -> String) :
-    BaseAdapter<MessageModel, LayoutMessageItemBinding, ChatAdapter.ChatViewHolder>(listener) {
+    private var click: ((MessageUIModel) -> Unit)? = null
+
+    fun onItemClick(block: (MessageUIModel) -> Unit) {
+        click = block
+    }
+
```

Comment:
ჩემი აზრით კლიკი უნდა გაიტანო onCreateViewHolder იმიტორო ყოველ ჯერზე არ დაისეტოს
https://stackoverflow.com/questions/28904479/best-place-to-attach-onclicklistener-in-recyclerview

Reply:
კი, makes sense to do that, მაგრამ onCreateViewHolder-ში ლისტიდან ვერ ვიღებ აითემს რომ პარამეტრად გავატანო ლამბდას